File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        stack<int>st;
        vector<int>ans(n,0);
        for(int i = 0; i < n; i++){
            while(!st.empty() && temperatures[i] > temperatures[st.top()]){
                int previdx = st.top(); st.pop(); // remove the top element
                ans[previdx] = i - previdx;
            }
            st.push(i); // new element push into the stack
        }
        return ans;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] result = new int[n];
        Arrays.fill(result, 0);
        Stack<Integer>st = new Stack<>();

        for(int i = 0; i < n; i++){
            while(!st.isEmpty() && temperatures[i] > temperatures[st.peek()]){
                int prev_idx = st.pop();
                result[prev_idx] = i - prev_idx;
            }
            st.push(i);
        }
        return result;
    }
}